src/mongocxx/test/spec/change_stream.cpp

```diff
@@ -41,79 +35,96 @@ using namespace bsoncxx;
+    } catch (const operation_exception& e) {
+    }
+
+    // Steps 2 - 5, set up new collection
+    set_up_collection(client, test_spec);
+    if (test_spec["database2_name"]) {
+        set_up_collection(client, test_spec, "database2_name", "collection2_name");
+    }
+
+    // Step 6. "If failPoint is specified, its value is a configureFailPoint command"
+    configure_fail_point(client, test);
+}
 
 void run_change_stream_tests_in_file(const std::string& test_path) {
     INFO("Test path: " << test_path);
     auto test_spec_value = test_util::parse_test_file(test_path);
     REQUIRE(test_spec_value);
     auto test_spec = test_spec_value->view();
 
-    client global_client(uri{});
-
-    std::string db1_name = to_string(test_spec["database_name"].get_utf8().value);
-    std::string db2_name = to_string(test_spec["database2_name"].get_utf8().value);
-    std::string coll1_name = to_string(test_spec["collection_name"].get_utf8().value);
-    std::string coll2_name = to_string(test_spec["collection2_name"].get_utf8().value);
     auto tests = test_spec["tests"].get_array().value;
 
     // This follows the sketch laid out in the change stream spec tests readme:
     // https://github.com/mongodb/specifications/tree/master/source/change-streams/tests#spec-test-runner
     for (auto&& test_el : tests) {
         auto test = test_el.get_document().value;
-
-        if (should_skip_spec_test(global_client, test)) {
-            continue;
-        }
-
-        // "Use globalClient to [drop and recreate the collections]".
-        global_client[db1_name].drop();
-        global_client[db2_name].drop();
-        {
-            using namespace bsoncxx::builder::basic;
-            global_client[db1_name][coll1_name].insert_one(make_document());
-            global_client[db2_name][coll2_name].insert_one(make_document());
-        }
-
-        // "Create a new MongoClient `client`"
-        spec::apm_checker apm_checker;
-        options::client client_opts;
-        // "Begin monitoring all APM events for `client`"
-        client_opts.apm_opts(apm_checker.get_apm_opts());
-        class client client(uri{}, client_opts);
-
-        INFO("Test description: " << to_string(test["description"].get_utf8().value));
-        if (test["description"].get_utf8().value.compare(
-                "Change Stream should error when an invalid aggregation stage is passed in") == 0) {
-            WARN("Skipping test with invalid pipeline stages. The C++ driver cannot test them.");
-            continue;
-        }
-
-        options::change_stream cs_opts{};
-        if (test["changeStreamOptions"]) {
-            auto options = test["changeStreamOptions"].get_document().value;
-            if (options["batchSize"]) {
-                cs_opts.batch_size(options["batchSize"].get_int32().value);
+        auto description = test["description"].get_utf8().value;
+        SECTION(to_string(description)) {
```

Comment:
Wrapping this in a `SECTION` shows the description under `TEST_CASE` ("Change stream spec tests") for failed tests. Which gives output like:
```
Change stream spec tests
  Executing a watch helper on a Collection results in notifications for changes
  to the specified collection
``` 
The descriptions of additional test cases that failed are displayed, indented, under the `TEST_CASE` as well.